price-feeder/oracle/oracle.go

```diff
@@ -393,6 +407,66 @@ func (o *Oracle) filterDeviations(
 	return filteredPrices, nil
 }
 
+// filterCandleDeviations finds the standard deviations of the tvwaps of
+// all assets, and filters out any providers that are not within 2𝜎 of the mean.
+func (o *Oracle) filterCandleDeviations(
+	candles provider.AggregatedProviderCandles,
+) (provider.AggregatedProviderCandles, error) {
+	var (
+		filteredCandles = make(provider.AggregatedProviderCandles)
+		tvwaps          = make(map[string]map[string]sdk.Dec)
+	)
+
+	for providerName, c := range candles {
+		candlePrices := make(provider.AggregatedProviderCandles)
+
+		for assetName, asset := range c {
+			if _, ok := candlePrices[providerName]; !ok {
+				candlePrices[providerName] = make(map[string][]provider.CandlePrice)
+			}
+			candlePrices[providerName][assetName] = asset
+		}
+
+		tvwap, err := ComputeTVWAP(candlePrices)
```

Comment:
So we call `ComputeTVWAP` when filtering and also after we've filtered?

Reply:
Yup, we calculate tvwap for each provider, use that to average & filter out, then calculate tvwap for all providers together - it's not that efficient but seemed the most clean, any ideas for how to change it?